### ISSUE TITLE
Refactor token without https error

### ISSUE THREAD

**Issue by juanignaciosl** [CartoDB/cartodb.js#449]
When [this check is triggered](https://github.com/CartoDB/cartodb.js/blob/101959f7e5d7a89391a61995c27dc407cca47bff/src/geo/layer_definition.js#L759) it logs an error but page keeps loading, so new errors can be generated (see  CartoDB/cartodb#3307) which are no obviously related (well, at least, not for me :_) ). Maybe it could be more explicit, display a user error or stop further processing, not sure.

**Comment by alonsogarciapablo** [comment]
I've been taking a look at this. I've been able to reproduce it locally, when serving the app via `http`. That [check](https://github.com/CartoDB/cartodb.js/blob/101959f7e5d7a89391a61995c27dc407cca47bff/src/geo/layer_definition.js#L758) is preventing the layer to be added to the mapView, and we're assuming that it's been added later in the code. Also, **I've searched for this error in track.js, and I haven't found any occurrence**.

@javisantana Will this ever happen in a production environment? Why is `https` required?

From what I understand, the access token is being used in the url to fetch the attributes that we then use to update the content of the infowindows when they are open... Instead of throwing this error and preventing the layer from being rendered, perhaps we could prevent the infowindows from "working" as expected when attributes can't be fetched...
